substrate/frame/support/procedural/src/lib.rs

```diff
@@ -476,6 +477,35 @@ pub fn storage_alias(attributes: TokenStream, input: TokenStream) -> TokenStream
 		.into()
 }
 
+/// Derive macro for simplifying storage type definitions.
+///
+/// This macro automatically applies the appropriate derives and bounds for types stored in
+/// blockchain storage. It handles phantom type parameters and `MaxEncodedLen` requirements.
+///
+/// # Arguments
+///
+/// * `skip(T, U, ...)` - Generic parameters that should not have bounds added (phantom types)
```

Comment:
This is not really about phantom types, it is about type that are used indirectly.

e.g. for `struct Vote<T>(Vec<T>)` what we actually want to bound if `Vec<T>` and not `T`. but rust standard macros are too naive in their implementation. codec macro handle it correctly.